### ISSUE TITLE
Start process doesn't gets completed in windows

### ISSUE THREAD

**Issue by hchockarprasad** [theseus-rs/postgresql-embedded#99]
I think pg_ctl start command waits forever during startup

```
let mut db = PostgreSQL::new(settings);
db.setup().await.unwrap();
println!("setup complete");
db.start().await.unwrap();
println!("Never reached");
```

**Comment by brianheineman** [comment]
Hello @hchockarprasad thank you for the bug report.  I was able to reproduce the issue and correct it within my environment.  The fix has been released in [0.14.1](https://github.com/theseus-rs/postgresql-embedded/releases/tag/v0.14.1).

**Comment by JMBeresford** [comment]
@brianheineman Seems this may still be an issue. The `start` command times out unconditionally for me, but only on Windows. Debug logs w/ `settings.timeout == 30s`:

```
  DEBUG  Starting embedded database: Settings {
    releases_url: "https://github.com/theseus-rs/postgresql-binaries",
    version: VersionReq {
        comparators: [
            Comparator {
                op: Exact,
                major: 17,
                minor: Some(
                    2,
                ),
                patch: Some(
                    0,
                ),
                pre: Prerelease(""),
            },
        ],
    },
    installation_dir: "C:\\Users\\volcaus\\.theseus\\postgresql",
    password_file: "C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\.passwd",
    data_dir: "C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\data",
    host: "localhost",
    port: 0,
    username: "postgres",
    password: "postgres",
    temporary: false,
    timeout: Some(
        30s,
    ),
    configuration: {},
}
    at packages\db\src\embedded.rs:14
    in start_embedded_db with url: "postgres://postgres:postgres@localhost/retrom?data_dir=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\data&password_file=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\.passwd"
    in get_server with db_params: Some("?data_dir=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\data&password_file=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\.passwd")
    in start_server
    in ipc::request::run
    in wry::ipc::handle

  DEBUG  Starting database C:\Users\volcaus\AppData\Roaming\com.retrom.client\db\data on port 53245
    at C:\Users\runneradmin/.cargo\registry\src\index.crates.io-6f17d22bba15001f\postgresql_embedded-0.17.3\src\postgresql.rs:229
    in start
    in start_embedded_db with url: "postgres://postgres:postgres@localhost/retrom?data_dir=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\data&password_file=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\.passwd"
    in get_server with db_params: Some("?data_dir=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\data&password_file=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\.passwd")
    in start_server
    in ipc::request::run
    in wry::ipc::handle

  DEBUG  Executing command: "C:\\Users\\volcaus\\.theseus\\postgresql\\17.2.0\\bin\\pg_ctl" "start" "--pgdata" "C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\data" "--wait" "--log" "C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\data\\start.log" "-o" "-F -p 53245"
    at C:\Users\runneradmin/.cargo\registry\src\index.crates.io-6f17d22bba15001f\postgresql_commands-0.17.3\src\traits.rs:185
    in execute_command with program: "pg_ctl"
    in start
    in start_embedded_db with url: "postgres://postgres:postgres@localhost/retrom?data_dir=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\data&password_file=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\.passwd"
    in get_server with db_params: Some("?data_dir=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\data&password_file=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\.passwd")
    in start_server
    in ipc::request::run
    in wry::ipc::handle

  ERROR  Database start error: TimeoutError(
    Elapsed(
        (),
    ),
)
    at packages\db\src\embedded.rs:39
    in start_embedded_db with url: "postgres://postgres:postgres@localhost/retrom?data_dir=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\data&password_file=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\.passwd"
    in get_server with db_params: Some("?data_dir=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\data&password_file=C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\.passwd")
    in start_server
    in ipc::request::run
    in wry::ipc::handle
```

When running the command directly, it succeeds nearly instantly:
```
PS C:\Users\volcaus\AppData\Local\Retrom> C:\\Users\\volcaus\\.theseus\\postgresql\\17.2.0\\bin\\pg_ctl start --pgdata C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\data --wait --log C:\\Users\\volcaus\\AppData\\Roaming\\com.retrom.client\\db\\data\\start.log -o "-F -p 52174"
waiting for server to start.... done
server started
```

**Comment by guitarhero22** [comment]
I'm getting the same error when trying to setup and use the server.
Was able to fixit by using tokio::process::Command::spawn() instead of tokio::process::Command::output(), in the impl of AsyncCommandExecutor trait.
I'm new to rust, coming from C++. Please bear with me if I'm not correctly using the scopes.

**Comment by brianheineman** [comment]
@JMBeresford and @guitarhero22  thank you for reporting the issue.  I'm not certain why this does not work for `pg_ctl` on Windows and may need to run the issue up to someone on the Rust project.  @guitarhero22 since you indicated this is working for you with `spawn`()` I will update the async executor to use that as well (the sync executor already does this).  Hopefully, this will address the issue you are seeing.

**Comment by brianheineman** [comment]
@JMBeresford and @guitarhero22 I just released [0.17.4](https://github.com/theseus-rs/postgresql-embedded/releases/tag/v0.17.4) with the change for `pg_ctl` on Windows when using the async command executor.  Please let me know if this addresses the issue for you.
